spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskLogsController.java

```diff
@@ -51,13 +51,23 @@ public TaskLogsController(TaskExecutionService taskExecutionService) {
 
 	/**
 	 * Retrieve logs for the task execution identified by the provided external execution ID
-	 * @param taskExternalExecutionId the external execution ID returned by the platform when launching the task
+	 * @param executionId the execution ID used to retrieve the log.
 	 * @param platformName the platform name
+	 * @param idType identifies whether the log should be obtained using the {@code externalExecutionId} or the {@code taskExecutionId} created by SCDF.
+	 * The default is "external" which means the method will use the {@code externalExecutionId}.  If you wish to use the {@code taskExecutionId} created by SCDF then set {@code idType} to "internal".
 	 * @return the log content represented as String
 	 */
-	@RequestMapping(value = "/{taskExternalExecutionId}", method = RequestMethod.GET)
+	@RequestMapping(value = "/{executionId}", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
-	public ResponseEntity<String> getLog(@PathVariable String taskExternalExecutionId, @RequestParam(required = false, defaultValue = "default") String platformName) {
-		return new ResponseEntity<>(this.taskExecutionService.getLog(platformName, taskExternalExecutionId), HttpStatus.OK);
+	public ResponseEntity<String> getLog(@PathVariable String executionId, @RequestParam(required = false, defaultValue = "default") String platformName, @RequestParam(required = false, defaultValue = "external") String idType) {
```

Comment:
Renaming the params will be breaking change to the API. We just ran into this w/ another PR and what we did was leave the existing badly named param and mark it is as deprecated and then add the new param and just use one of them. Its messy but I am not sure of a better way.